src/omnicore/omnicore.cpp

```diff
@@ -2304,86 +2304,66 @@ int mastercore::ClassAgnosticWalletTXBuilder(const string &senderAddress, const
+           // is it a valid alert?
+            if (atoi(vstr[2]) == OMNICORE_MESSAGE_TYPE_ALERT && atoi(vstr[0]) == 1) {
+                // is it the most recent and within our parsed range?
+                if (atoi(vstr[1]) > lastAlertBlock && atoi(vstr[1]) <= blockHeight) {
+                    lastAlertTxid = skey.ToString();
+                    lastAlertData = svalue.ToString();
+                    lastAlertBlock = atoi(vstr[1]);
+                }
            }
        }
     }
     delete it;
 
     // if lastAlertTxid is not empty, load the alert and see if it's still valid - if so, copy to global_alert_message
-    if(lastAlertTxid.empty())
-    {
+    if (lastAlertTxid.empty()) {
         PrintToLog("DEBUG ALERT No alerts found to load\n");
+        return false;
     }
-    else
-    {
-        PrintToLog("DEBUG ALERT Loading lastAlertTxid %s\n", lastAlertTxid);
-
-        // reparse lastAlertTxid
-        uint256 hash;
-        hash.SetHex(lastAlertTxid);
-        CTransaction wtx;
-        uint256 blockHash = 0;
-        if (!GetTransaction(hash, wtx, blockHash, true)) //can't find transaction
-        {
-            PrintToLog("DEBUG ALERT Unable to load lastAlertTxid, transaction not found\n");
-        }
-        else // note reparsing here is unavoidable because we've only loaded a txid and have no other alert info stored
-        {
-            CMPTransaction mp_obj;
-            int parseRC = ParseTransaction(wtx, blockHeight, 0, mp_obj);
-            string new_global_alert_message;
-            if (0 <= parseRC)
-            {
-                if (0<=mp_obj.step1())
-                {
-                    if(65535 == mp_obj.getType())
-                    {
-                        // TODO: use new parsing/interpretation functions
-                        if (0 == mp_obj.step2_Alert(&new_global_alert_message))
-                        {
-                            SetOmniCoreAlert(new_global_alert_message);
-                            // check if expired
-                            CBlockIndex* pBlockIndex = chainActive[blockHeight];
-                            if (pBlockIndex != NULL) {
-                                CheckExpiredAlerts(blockHeight, pBlockIndex->GetBlockTime());
-                            }
-                        }
-                    }
-                }
+
+    PrintToLog("DEBUG ALERT Loading lastAlertTxid %s\n", lastAlertTxid);
+
+    // reparse the last alert
+    uint256 hash(lastAlertTxid);
+    CTransaction tx;
+    uint256 blockHash;
+    if (!GetTransaction(hash, tx, blockHash, true)) {
+        PrintToLog("DEBUG ALERT Unable to load lastAlertTxid, transaction not found\n");
+        return false;
+    }
+
+    CMPTransaction alertTx;
+    if (ParseTransaction(tx, blockHeight, 0, alertTx) == 0) {
+        // TODO: calling interpretPacket, and logic in general, outside of the
+        // regular flow seems dangerous
+        if (alertTx.interpretPacket() == 0) {
```

Comment:
Well, I hope this didn't sound too rude, hehe. But I'm basically thinking: if "they" don't care, then they should be burned [to a reasonable degree] and face the consequences. If a consequence is one day downtime, because we fired an alert as last resort to outdated clients, then that's fine.

In such a case there are probably two outcomes:
 1. the Omni Layer isn't going to be supported in the future
 2. more care is taken

I'd like to avoid (1) by all means, and do as much as possible to encourage (2).